build-support/bin/ci.py

```diff
@@ -175,30 +196,85 @@ def get_remote_execution_oauth_token_path() -> Iterator[str]:
     yield tf.name
 
 # -------------------------------------------------------------------------
-# Blacklists
+# Blacklists and whitelists
 # -------------------------------------------------------------------------
 
-def get_blacklisted_targets(filename: str) -> Set[str]:
-  return {
-    line.strip()
-    for line in Path(f"build-support/ci_blacklists/{filename}").read_text().splitlines()
-  }
+Target = str
+TargetSet = Set[Target]
 
 
-def get_all_python_tests(*, tag: Optional[str] = None) -> Set[str]:
-  command = [
-    "./pants.pex",
-    "--filter-type=python_tests",
-    "filter",
-    "src/python::",
-    "tests/python::",
-    "contrib::"
-  ]
-  if tag is not None:
-    command.insert(1, f"--tag={tag}")
-  return set(subprocess.run(
-    command, stdout=subprocess.PIPE, encoding="utf-8", check=True
-  ).stdout.strip().split("\n"))
+class DefaultTestBehavior(Enum):
+  v1_no_chroot = 1
+  v2_remote = 2
+
+
+class TestType(Enum):
+  unit = "unit"
+  integration = "integration"
+
+  def __str__(self):
+    return str(self.value)
+
+
+class TestTargetSets(NamedTuple):
+  v1_no_chroot: TargetSet
+  v1_chroot: TargetSet
+  v2_local: TargetSet
+  v2_remote: TargetSet
+
+  @staticmethod
+  def calculate(
+    *, test_type: TestType, default_behavior: DefaultTestBehavior, remote_execution_enabled: bool
+  ) -> 'TestTargetSets':
+
+    def get_listed_targets(filename: str) -> TargetSet:
+      list_path = Path(f"build-support/ci_lists/{filename}")
+      if not list_path.exists():
+        return set()
+      return {line.strip() for line in list_path.read_text().splitlines()}
+
+    all_targets = set(subprocess.run(
+      [
```

Comment:
nit: I'd probably extract this cli into a function (or a template string variable), like you have done with `remote_execution_command`, but given that it happens only once, it's not a big deal.